modules/asm/variables.tf

```diff
@@ -64,7 +64,7 @@ variable "asm_git_tag" {
 }
 
 variable "mode" {
-  description = "ASM mode for deployment. Supported mode is `install` only."
+  description = "ASM mode for deployment. Supported modes are `install` and `upgrade`."
```

Comment:
Does this work as expected? When the user swaps mode from install to upgrade it will cause the [destroy](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/c32c5d162c5a7ca52a563ecf3e5557c3c4bd32ba/modules/asm/main.tf#L48) to be run and then create to be rerun in `upgrade` mode

Reply:
Although TF reports running destroy it is no-op as we [explicitly](https://github.com/terraform-google-modules/terraform-google-gcloud/blob/48e1fee9892f08dc71761d6ac4b06173c28a87fd/main.tf#L259) only activate it for `destroy` phase